### ISSUE TITLE
UPNP client does not work on wireless only

### ISSUE THREAD

**Issue by travisghansen** [OpenELEC/OpenELEC.tv#4558]
I've got an RPi2 with openelec-6.0.0 installed.  When trying to add a UPNP source nothing is found in the list.  If I physically plug in with a wired connection it all works as expected.  My guess is conman (or whatever is used) has some different default firewall rules for wireless vs wired connections.

Thanks!

**Comment by lrusak** [comment]
Nope, most likely it is your router blocking multicast. There is no firewall (other than iptables, but with no default rules) installed in OpenELEC.

**Comment by travisghansen** [comment]
@lrusak certainly not my router as I have many UPnP clients over wireless working fine.  I'll try the ifconfig commands and report back however..

**Comment by travisghansen** [comment]
@lrusak OK, so here's what I've found:

1. multicast is enabled fine
2. connectivity is fine
3. it appears to be a timing issue

If I tell elec to 'Wait for network before starting Kodi' then all works as expected.  So apparently if a connection is made after Kodi starts up things can't get 'refreshed' as no matter how long elec is up and running after a boot without the 'wait' setting enabled the devices never appear.

Any other info I can provide?

**Comment by tspoeri** [comment]
I can confirm this behavior on RPi2 with openelec-6.0.1

**Comment by chewitt** [comment]
@travisghansen - please test a current milhouse build that contains "connman-wait-online" changes to see if this resolves the timing issue? - Thanks (and Hello!)

**Comment by travisghansen** [comment]
@chewitt Hey!  With the milhouse build am I supposed to uncheck the "Wait for network" setting and retest?

**Comment by MilhouseVH** [comment]
@travisghansen : Try with "Wait for network" enabled and then disabled, does either work?

**Comment by travisghansen** [comment]
@MilhouseVH OK, but as stated above the current production builds already work with it enabled.  Just wanted to make sure everyone's clear.
